pkg/rdt/rdt.go

```diff
@@ -87,6 +90,30 @@ type MonGroup interface {
 	Parent() CtrlGroup
 }
 
+// Monitor is the interface for retrieving RDT monitoring data
+type Monitor interface {
+	GetMonData() MonData
+}
+
+// MonData contains monitoring stats of one monitoring group
+type MonData struct {
+	L3 MonL3Data
+}
+
+// MonL3Data contains L3 monitoring stats of one monitoring group
+type MonL3Data map[uint64]MonLeafData
+
+// MonLeafData represents the raw numerical stats from one RDT monitor data leaf
+type MonLeafData map[string]uint64
+
+// MonResource is the type of RDT monitoring resource
+type MonResource string
+
+const (
+	// MonResourceL3 is the RDT L3 cache monitor resource
+	MonResourceL3 MonResource = "l3"
+)
```

Comment:
Will there be more (for memory bandwidth?), or what is the reason for defining a type with/for a single resource ?

Reply:
I really don't know. This construct was just to make the code more future-proof, as the kernel interface could easily be extended. We might get L2 monitoring in the future, for example. This is pure speculation, though.

Should I change this, WDYT?

Reply:
No, don't change it.